diff --git a/components/bin/build b/components/bin/build
index 68360eac4..ef46205c7 100755
--- a/components/bin/build
+++ b/components/bin/build
@@ -61,6 +61,9 @@ process.chdir(path.dirname(json));
 const mjPath = path.relative(process.cwd(), path.resolve(__dirname, '..', '..', target));
 const mjGlobal = path.join('..', mjPath, 'components', 'global.js');
 
+/**
+ * Determine the module type
+ */
 function getType() {
   const component = config.component || 'part';
   if (component.match(/\/(svg|chtml|common)\/fonts\//)) return RegExp.$1 + '-font';
@@ -69,6 +72,13 @@ function getType() {
   return component;
 }
 
+/**
+ * Convert Windows paths to unix paths
+ */
+const normalize = process.platform === 'win32'
+      ? (file) => file.replace(/\\/g, '/')
+      : (file) => file;
+
 /**
  * Extract the configuration values
  */
@@ -100,7 +110,7 @@ let PACKAGE = [];
  */
 function processList(base, dir, list, top = true) {
   for (const item of list) {
-    const file = path.join(dir, item);
+    const file = normalize(path.join(dir, item));
     if (!EXCLUDE.has(file)) {
       const stat = fs.statSync(path.resolve(base, file));
       if (stat.isDirectory()) {
@@ -271,12 +281,12 @@ function getExtraDirectories() {
 function processGlobal() {
   console.info('  ' + COMPONENT + '.ts');
   const lines = (target === 'cjs' ? [
-    `const {combineWithMathJax} = require('${GLOBAL}')`,
-    `const {VERSION} = require('${VERSION}');`,
+    `const {combineWithMathJax} = require('${normalize(GLOBAL)}')`,
+    `const {VERSION} = require('${normalize(VERSION)}');`,
     '',
   ] : [
-    `import {combineWithMathJax} from '${GLOBAL}';`,
-    `import {VERSION} from '${VERSION}';`,
+    `import {combineWithMathJax} from '${normalize(GLOBAL)}';`,
+    `import {VERSION} from '${normalize(VERSION)}';`,
     '',
   ]);
   const [prefix, indent, postfix] = getExtraDirectories();
@@ -337,7 +347,7 @@ function processPackage(lines, space, dir) {
     if (path.dirname(PACKAGE[0]) === dir) {
       const file = PACKAGE.shift();
       const name = path.basename(file);
-      let relativefile = path.join('..', JS, dir, name).replace(/\.ts$/, '.js')
+      const relativefile = normalize(path.join('..', JS, dir, name).replace(/\.ts$/, '.js'));
       const component = 'module' + (++importCount);
       lines.push(
         target === 'cjs' ?
diff --git a/components/bin/makeAll b/components/bin/makeAll
index b661d3696..8eafa678c 100755
--- a/components/bin/makeAll
+++ b/components/bin/makeAll
@@ -124,6 +124,13 @@ function fileRegExp(name) {
   return new RegExp(name.replace(/([\\.{}[\]()?*^$])/g, '\\$1'), 'g');
 }
 
+/**
+ * Options for the execSync() function
+ */
+const execOptions = process.platform === 'win32'
+      ? { shell: `${process.env.ProgramFiles}\\Git\\bin\\bash.exe` }
+      : {};
+
 /**
  * Get the current working directory
  */
@@ -196,7 +203,7 @@ function processSubdirs(dir, action, config) {
  * Run a command on a given directory
  */
 function run(cmd, dir) {
-  return execSync(cmd + ` '${path.relative('.', dir).replace(/'/g, '\\\'')}'`);
+  return execSync(cmd + ` '${path.relative('.', dir).replace(/'/g, '\\\'')}'`, execOptions);
 }
 
 /**
diff --git a/components/bin/pack b/components/bin/pack
index ebcf8a214..4a91ab5de 100755
--- a/components/bin/pack
+++ b/components/bin/pack
@@ -25,7 +25,7 @@
 
 const fs = require('fs');
 const path = require('path');
-const {spawn, execSync} = require('child_process');
+const {spawn} = require('child_process');
 
 /**
  * The module type to use ('cjs' or 'mjs')
@@ -64,14 +64,7 @@ const rootRE = fileRegExp(path.dirname(jsPath));
 const nodeRE = /^.*\/node_modules/;
 const fontRE = new RegExp('^.*\\/(mathjax-[^\/-]*)(?:-font)?\/(build|[cm]js)');
 
-/**
- * Find the directory where npx runs (so we know where "npx webpack" will run)
- * (We use npx rather than pnpm here as it seems that pnpm doesn't
- * find the executable from a node_modules directory higher than the
- * first package.json, and extensions and fonts can have their own
- * package.json.)
- */
-const packDir = String(execSync('npx node -e "console.log(process.cwd())"'));
+const packDir = process.cwd();
 
 /**
  * @param {string} dir   The directory to pack
@@ -80,11 +73,20 @@ const packDir = String(execSync('npx node -e "console.log(process.cwd())"'));
 async function readJSON(dir) {
   return new Promise((ok, fail) => {
     const buffer = [];
-    const child = spawn('npx', [
-      'webpack', '--env', `dir=${path.relative(packDir, path.resolve(dir))}`,
-      '--env', `bundle=${bundle}`, '--json',
-      '-c', path.relative(packDir, path.join(compPath, 'webpack.config.' + target))
-    ]);
+    const child = spawn(
+      'npx',
+      [
+        'webpack',
+        '--env', `dir=${path.relative(packDir, path.resolve(dir))}`,
+        '--env', `bundle=${bundle}`,
+        '--json',
+        '-c', path.relative(packDir, path.join(compPath, 'webpack.config.' + target))
+      ],
+      {
+        cwd: packDir,
+        shell: true,
+      }
+    );
     child.stdout.on('data', (data) => buffer.push(String(data)));
     child.stderr.on('data', (data) => console.error(String(data)));
     child.on('close', (code) => {
diff --git a/package.json b/package.json
index c5a1fe05b..907d0d2f0 100644
--- a/package.json
+++ b/package.json
@@ -86,9 +86,9 @@
     "copy:mml3": "copy() { pnpm -s log:single 'Copying legacy code MathML3'; pnpm copyfiles -u 1 ts/input/mathml/mml3/mml3.sef.json $1; }; copy",
     "copy:pkg": "copy() { pnpm -s log:single \"Copying package.json to $1\"; pnpm copyfiles -u 2 components/bin/package.json $1; }; copy",
     "=============================================================================== log": "",
-    "log:comp": "log() { echo \\\\033[32m$1\\\\033[0m; }; log",
-    "log:header": "log() { echo '============='; echo $1; echo '============='; }; log",
-    "log:single": "log() { echo \\\\033[34m--$1\\\\033[0m; }; log",
+    "log:comp": "log() { echo \u001B[32m$1\u001B[0m; }; log",
+    "log:header": "log() { echo '\u001B[1m============='; echo $1; echo '=============\u001B[0m'; }; log",
+    "log:single": "log() { echo \u001B[94m--$1\u001B[0m; }; log",
     "=============================================================================== cjs": "",
     "cjs:build": "pnpm -s log:header 'Building cjs'; pnpm -s cjs:src:build && pnpm -s cjs:components:build",
     "cjs:bundle:clean": "pnpm clean:dir bundle-cjs",
